### PR TITLE
More search bugs

### PR DESCRIPTION
**Pull request by hamburge**
:bug: :bee: :beetle: :snail: :ant: 

-#1929 Put search shortcut in the Keyboard Shortcuts dialog and moved the command to `Window` from `Layer`, but if the executive decision is to put it in `Edit` (or somewhere else), I can move it again.

-I was also dissatisfied with how things looked after 507b561bf74b2acfe2b800472121b2013010b704, since the text would sometimes highlight momentarily before going to the end of the word, so I undid most of those changes and added a prop to TextInput for when we will never be selecting all text. I think it now looks cleaner and means there are fewer changes in `TextInput.jsx`

-Also #1930, #1931, #1932

### CODE REVIEW

**Comment by iwehrman** [comment]
@placegraphichere: do you approve of the placement of the `Search` menu command in the `Window` menu? Another plausible alternative might be the `Edit` menu.

**Comment by iwehrman** [comment]
Also, generally speaking, it is preferable to have smaller pull requests compared to larger ones. Next time, if it's possible, it would be nice to have one PR per issue. Sometimes that's not possible, or it's really inconvenient, and that's fine. But smaller PRs tend to be much easier to review, which means there tends to be a much shorter turnaround time before merging!

**Comment by iwehrman** [comment]
And, finally, needs a rebase!

**Comment by hamburge** [comment]
@iwehrman Just rebased-- not sure how some of your commits got in here?

**Comment by hamburge** [comment]
^jk, fixed that

File: src/nls/root/strings.js [hunk 1]
```diff
@@ -63,6 +63,216 @@ define(function (require, exports, module) {
             UNGROUP_LAYERS: "Ungroup layers"
         },
         APP_NAME: "Photoshop",
+        MENU: {
```
Comment (iwehrman) [inline src/nls/root/strings.js]: This is a rebase error. There is no longer a `MENU` property in strings.js. Instead, this has all moved to `i18n!nls/menu`.

**Comment by iwehrman** [comment]
Besides the rebase error, this looks good. I'm going to file a separate issue about the use of string encoding to identify filters, etc. Back to @hamburge to fix the menu stuff.

**Comment by iwehrman** [comment]
Opened #1966 for the aforementioned string encoding cleanup request.

**Comment by placegraphichere** [comment]
We’re using CMD+f which is the same as "find" for most apps. That is generally housed in the Edit menu. Usually that “help” or search finder located in the window menu is meant for it’s own OS specific application search capability (at least on mac). That kind of search is definitely different than ours. 

So for this case, I might lean towards Edit as a home. That way we can also break it out into more categorical searches in the future, say for instance *Edit > Search > Stock* or *Edit > Search > Libraries*. Housing them under Window would imply (to me anyway) it will take me out of the app to the web.

**Comment by hamburge** [comment]
@iwehrman Great, thanks. I moved the menu command over to Edit as well.

**Comment by iwehrman** [comment]
OK!

**Pull request merged**